python/tests/sql_aggregates/test_average.py

```diff
@@ -0,0 +1,183 @@
+            {"c1": 11, "c2": 22, "c3": None, "c4": 45, "c5": 51, "c6": 6, "c7": 72, "c8": 88},
+            {"c1": 14, "c2": 21, "c3": 44, "c4": 32, "c5": 34, "c6": 90, "c7": 29, "c8": 67}]
+        self.add_data(new_data)
+        expected_data = [{"c1": 8, "c2": 15, "c3" : 25, "c4" : 27, "c5" : 30, "c6" : 52, "c7": 57, "c8": 54}]
+        table_name = "avg_distinct"
+        view_query = f'''CREATE VIEW avg_view AS SELECT
+                            AVG(DISTINCT c1) AS c1, AVG(DISTINCT c2) AS c2, AVG(DISTINCT c3) AS c3, AVG(DISTINCT c4) AS c4, AVG(DISTINCT c5) AS c5, AVG(DISTINCT c6) AS c6, AVG(DISTINCT c7) AS c7, AVG(DISTINCT c8) AS c8
+                        FROM {table_name}'''   
+        self.execute_query(pipeline_name, expected_data, table_name, view_query)   
+
+class Avg_Distinct_Groupby(TestAggregatesBase):
+    def test_avg_distinct_groupby(self):
+        pipeline_name = "test_distinct_groupby"
+        # validated using postgres
+        new_data = [
+            {"id" :0 ,"c1": None, "c2": 2, "c3": 30, "c4": 4, "c5": None, "c6": 60, "c7": 70, "c8": 8},
+            {"id": 1,"c1": 11, "c2": 22, "c3": None, "c4": 45, "c5": 51, "c6": 6, "c7": 72, "c8": 88},
+            {"id": 1,"c1": 14, "c2": 21, "c3": 44, "c4": 32, "c5": 34, "c6": 90, "c7": 29, "c8": 67}]
+        self.add_data(new_data)
+        expected_data = [
+            {"id": 0, "c1": 1, "c2": 2, "c3": 16, "c4": 4, "c5": 5, "c6": 33, "c7": 70, "c8": 8},
+            {"id" : 1, "c1": 12, "c2": 21, "c3" : 44, "c4" : 38, "c5" : 42, "c6" : 48, "c7":50, "c8": 77}]
+        table_name = "avg_distinct_groupby"
+        view_query = f'''CREATE VIEW avg_view AS SELECT
+                            id, AVG(DISTINCT c1) AS c1, AVG(DISTINCT c2) AS c2, AVG(DISTINCT c3) AS c3, AVG(DISTINCT c4) AS c4, AVG(DISTINCT c5) AS c5, AVG(DISTINCT c6) AS c6, AVG(DISTINCT c7) AS c7, AVG(DISTINCT c8) AS c8
+                         FROM {table_name}
+                         GROUP BY id;'''
+        self.execute_query(pipeline_name, expected_data, table_name, view_query) 
+
+class Avg_Where(TestAggregatesBase):
+    def test_avg_where0(self):
+        pipeline_name ="test_avg_where0"
+        # validated using postgres
+        new_data = [
+            {"c1": None, "c2": 2, "c3": 30, "c4": 4, "c5": None, "c6": 60, "c7": 70, "c8": 8},
+            {"c1": 11, "c2": 22, "c3": None, "c4": 45, "c5": 51, "c6": 6, "c7": 72, "c8": 88},
+            {"c1": 14, "c2": 21, "c3": 44, "c4": 32, "c5": 34, "c6": 90, "c7": 29, "c8": 67}]
+        self.add_data(new_data)
+        expected_data = [{"c1": 14, "c2": 21, "c3" : 44, "c4" : 32, "c5" : 34, "c6" : 90, "c7": 29, "c8": 67}]
+        table_name = "avg_where0"
+        view_query = f'''CREATE VIEW avg_view AS SELECT
+                            AVG(c1) AS c1, AVG(c2) AS c2, AVG(c3) AS c3, AVG(c4) AS c4, AVG(c5) AS c5, AVG(c6) AS c6, AVG(c7) AS c7, AVG(c8) AS c8
+                        FROM {table_name}
+                        WHERE 
```

Comment:
the WHERE next to the AVG, as above
this is true for all tests, I will stop repeating it